### ISSUE TITLE
Not enough Mining missions 

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#708]
**Describe the bug**
- On sol 19,  a total of 45 missions have started or completed in a default 9-settlement scenario. Only one Mining has started/completed. 

**Expected and unexpected behavior**
- There should have been one Mining mission as a follow-up for every 2 or 3 Exploration Mission

**Screenshots**
The only Mining mission is this one on row 30
![image](https://user-images.githubusercontent.com/1168584/193341257-a2d22ff6-4ddd-4363-9b01-4fd92b315320.png)

According to the phase log, it attempted to mine a site at 01-Adir-12:655
However, after 210 millisols, namely at 865 millisols, it quits before of "EVA - Not enough sunlight". 
![image](https://user-images.githubusercontent.com/1168584/193360297-b92ec2d7-49cc-4538-a66b-7a4b17961d05.png)

At 891 millisols on the same day, it starts traveling again.
![image](https://user-images.githubusercontent.com/1168584/193360608-e1cd1d73-2876-427c-955a-05b88591c2d1.png)

**Stacktrace**
```
01-Adir-09:622.147 (Info) AbstractMission : [New Plymouth] Stephen Garcia - Began organizing a Mining #32 mission.
01-Adir-09:622.147 (Info) AbstractVehicleMission : [New Plymouth] Stephen Garcia - Started mission Mining #32 using Gemini
01-Adir-09:623.259 (Info) Read : [38.3543 N 169.2236 E - Roadster] Elie Thibault - Started reading on reporting.
01-Adir-09:623.259 (Info) PlanMission : [New Plymouth] Stephen Garcia - Submitted a mission plan for Mining #32.
01-Adir-09:623.259 (Info) MissionManager : [New Plymouth] Stephen Garcia - Put together a mission plan for Mining #32.
01-Adir-09:623.259 (Config) MissionManager : Added 'Mining #32' mission.
...
01-Adir-09:727.744 (Info) ReviewMissionPlan : [New Plymouth] Crystal Lee - Grading Stephen Garcia's Mining #32 plan -  Rating: 10.0, Rels: 16, Quals: 11.25, Obj: 0.0, Emer: 0, Site: 10.0, Dist: -8, Lead: 7, Review: 10, Luck: 0 = Subtotal: 56.3
...
01-Adir-10:004.146 (Config) MasterClock :  - - - - - - - - - - - - - - Sol 10 - - - - - - - - - - - - - - 
01-Adir-10:004.146 (Info) ReviewMissionPlan : [New Plymouth] Larry Taylor - Grading Stephen Garcia's Mining #32 plan -  Rating: 10.0, Rels: 3, Quals: 7.5, Obj: 0.0, Emer: 0, Site: 10.0, Dist: -8, Lead: 5, Review: 2, Luck: -4 = Subtotal: 25.5
...
01-Adir-10:011.390 (Info) ReviewMissionPlan : [New Plymouth] Gregory Lopez - Grading Stephen Garcia's Mining #32 plan -  Rating: 10.0, Rels: 9, Quals: 7.5, Obj: 0.0, Emer: 0, Site: 10.0, Dist: -8, Lead: 7, Review: 4, Luck: -3 = Subtotal: 36.5
01-Adir-10:011.390 (Info) ReviewMissionPlan : [New Plymouth] Gregory Lopez - Grading Stephen Garcia's Mining #32 plan -  Rating: 10.0, Rels: 9, Quals: 7.5, Obj: 0.0, Emer: 0, Site: 10.0, Dist: -8, Lead: 7, Review: 4, Luck: -1 = Subtotal: 38.5
...
01-Adir-10:050.359 (Info) ReviewMissionPlan : [New Plymouth] Larry Taylor - Grading Stephen Garcia's Mining #32 plan -  Rating: 10.0, Rels: 3, Quals: 7.5, Obj: 0.0, Emer: 0, Site: 10.0, Dist: -8, Lead: 5, Review: 2, Luck: 3 = Subtotal: 32.5
...
01-Adir-10:221.350 (Info) ReviewMissionPlan : [New Plymouth] Crystal Lee - Grading Stephen Garcia's Mining #32 plan -  Rating: 10.0, Rels: 16, Quals: 11.25, Obj: 0.0, Emer: 0, Site: 10.0, Dist: -8, Lead: 7, Review: 10, Luck: 4 = Subtotal: 60.3
...
01-Adir-10:225.916 (Info) ReviewMissionPlan : [New Plymouth] Crystal Lee - Approved Stephen Garcia's Mining #32 mission plan. Total Score: 462.0 [Min: 262.4].
...
01-Adir-10:291.287 (Info) AbstractMission : [New Plymouth] Stephen Garcia - Mission plan for Mining #32 was approved.
01-Adir-10:292.377 (Info) AbstractVehicleMission : [New Plymouth] Gemini - Prepared a loading plan sourced from New Plymouth.
01-Adir-10:292.377 (Info) TaskManager [x2] : [New Plymouth] Stephen Garcia - Given a new task order of 'LoadVehicleEVA'.
...
01-Adir-14:595.382 (Info) AbstractMission : [New Plymouth] Stephen Garcia - Ended the Mining #32 with the status flag(s): 
01-Adir-14:595.382 (Info) AbstractMission : [New Plymouth] Stephen Garcia - Disbanding mission member(s): Stephen Garcia, Crystal Lee, Gregory Lopez
```
**Specifications  (please complete the following information):**
 - pre-3.4.0 build 7437 
 
**Additional context**
- Schiaparelli Point is the largest and most resourceful settlement among the 9 settlements. However, it is still making new Exploration missions, as opposed to starting Mining mission. 
- What command do we need in the text Console or in Swing based UI to look into what's missing in automatically creating a Mining mission for each settlement ?

**Comment by bevans2000** [comment]
This is a duplicate of #628

**Comment by mokun** [comment]
@bevans2000 

Can you make it stay for another day to mine minerals ? 

This trip result in having no minerals excavated because it only stayed there for ~200 millisols -- not enough time to mine.

**Comment by bevans2000** [comment]
Sure a longer stay is easy. I can do it tomorrow but I think the problem will be the low probability score. The sites have to have been sufficiently explored before they can be mined. It is possible with the Console to artificially create fully explored sites then use the create mission command.

**Comment by mokun** [comment]
In this sim session, on sol 18, we have one on-going Mining mission. 

It departed from Starcity on sol 08:156

![image](https://user-images.githubusercontent.com/1168584/193428521-801de10e-d989-4e02-817e-716c3046cc1d.png)

It arrived at the mining site on sol 8:256 and had since been mining in the last 10 sols.
Currently it's on sol 18:492 and still at the mining site.

![image](https://user-images.githubusercontent.com/1168584/193428990-64789410-63db-4bbe-9308-f6f982ee7cf2.png)

Interestingly, the Vehicle Status said it's "Parked" and "Towing".

It turns out it's just towing `LUV 012` needed for mining operation.

![image](https://user-images.githubusercontent.com/1168584/193429035-c71b57dd-5cbb-4229-9275-b85ff653fbe4.png)

Regrettably, Keith Watson could not join in this mission. Also, the other 3 members boarded the vehicle are somehow busy at doing other things and not mining.

![image](https://user-images.githubusercontent.com/1168584/193429263-3fb3b6c3-5d85-4ce6-a946-2d57a7c10690.png)

They are very inefficient in what they do. 
After 10 sols, they excavated ZERO minerals.

Also, hematite has 1.9 % estimated and 0 % actual. The rest of the mineral also have 0 % actual. 

@bevans2000
1. We need to modify MineSite task to put them into real actions at the mining site.

2. Also, Odyssey a *Transport Rover* and it should have more occupants onboard to help out with mining operation. 

3. If one person doesn't show up, should they recruit another member as his replacement ?  

4. We could deviate the actual % to be different than the estimated % for each mineral.

May be we can create a certainty factor and this factor gets higher if they spent time analyzing more maps at this site. 

A high certainty factor would give them more assurance that the estimated % would be closer to actual %.

It seems in this case, this is NOT a good mining site at all.

**Comment by mokun** [comment]
Below is the result of my query on this sim session :

```
01-Adir-18:492 MarsLink >  ​/ep
                Location |           Settlement |   Status | Reviews | Highest
------------------------------------------------------------------------------
     16.8504 N 49.0956 W |               Europa |          |       0 | taenite - 19.98
     16.8312 N 49.1025 W |               Europa |          |       0 | olivine - 18.94
     16.7673 N 49.0797 W |               Europa |          |       0 | olivine - 18.79
     16.7505 N 49.1616 W |               Europa |          |       0 | malachite - 17.46
     16.6920 N 49.1968 W |               Europa |          |       0 | olivine - 17.88
    41.1399 N 162.0045 E |             Starcity |          |       0 | chalcopyrite - 19.30
    41.1608 N 161.9271 E |             Starcity |          |       0 | chalcopyrite - 21.27
    41.1148 N 161.8975 E |             Starcity |          |       0 | chalcopyrite - 31.31
    41.0955 N 161.9042 E |             Starcity |          |       0 | malachite - 19.98
    41.0496 N 161.8739 E |             Starcity |          |       0 | chalcopyrite - 27.50
    41.1398 N 162.0053 E |             Starcity |    Mined |      33 | chalcopyrite - 10.07
      9.3540 N 33.0414 W |         New Plymouth |          |       0 | sylvite - 19.64
      9.2979 N 33.1054 W |         New Plymouth |          |       0 | hematite - 27.74
      9.2954 N 33.1812 W |         New Plymouth |          |       0 | hematite - 43.95
      9.2250 N 33.2284 W |         New Plymouth |          |       0 | hematite - 36.16
      9.3787 N 33.1674 W |         New Plymouth |          |       0 | hematite - 21.33
    41.1605 N 161.9282 E |             Starcity | Explored |      65 | chalcopyrite - 10.32
    41.1154 N 161.8979 E |             Starcity | Explored |      41 | chalcopyrite - 13.29
     39.1695 N 99.9823 W |            New Seoul |          |       0 | magnesite - 17.13
     39.1849 N 99.8975 W |            New Seoul |          |       0 | kamacite - 17.00
     39.0741 N 99.8443 W |            New Seoul |          |       0 | malachite - 18.71
     39.0186 N 99.7947 W |            New Seoul |          |       0 | malachite - 18.44
     38.9567 N 99.7600 W |            New Seoul |          |       0 | goethite - 19.89
    41.1398 N 162.0067 E |             Starcity | Explored |      31 | chalcopyrite - 11.21
    41.0955 N 161.9042 E |             Starcity | Explored |      26 | chalcopyrite - 18.69
       1.8636 N 1.9140 E |   Schiaparelli Point |          |       0 | chalcopyrite - 18.53
       1.8663 N 1.9424 E |   Schiaparelli Point |          |       0 | sylvite - 16.52
       1.8419 N 1.9651 E |   Schiaparelli Point |          |       0 | goethite - 19.32
       1.7363 N 2.0493 E |   Schiaparelli Point |          |       0 | magnesite - 15.01
       2.0836 N 2.2011 E |   Schiaparelli Point |          |       0 | chalcopyrite - 18.38
    41.0510 N 161.8748 E |             Starcity |          |       2 | chalcopyrite - 32.89
    41.0961 N 161.9040 E |             Starcity | Explored |      55 | chalcopyrite - 14.27
     39.1701 N 99.9823 W |            New Seoul |          |      34 | magnesite - 0.00
    41.0500 N 161.8741 E |             Starcity | Explored |      84 | chalcopyrite - 17.30
       1.8635 N 1.9139 E |   Schiaparelli Point | Explored |      53 | magnesite - 0.00
      9.3542 N 33.0409 W |         New Plymouth | Explored |      20 | hematite - 16.86
    20.4151 N 131.1040 E |           Tian Cheng |          |       0 | kamacite - 18.03
      9.2986 N 33.1047 W |         New Plymouth | Explored |      11 | hematite - 26.53
    33.7969 N 146.3224 W |              Kyocera |          |       0 | malachite - 18.13
    33.8280 N 146.2502 W |              Kyocera |          |       0 | malachite - 19.19
    33.7220 N 146.1874 W |              Kyocera |          |       0 | taenite - 19.78
    33.6702 N 146.2395 W |              Kyocera |          |       0 | chalcopyrite - 17.50
    33.7313 N 146.3418 W |              Kyocera |          |       0 | olivine - 19.21
    41.1399 N 162.0047 E |             Starcity | Explored |      77 | chalcopyrite - 10.08
      9.2253 N 33.2281 W |         New Plymouth | Explored |      22 | hematite - 28.55
    41.1398 N 162.0063 E |             Starcity | Explored |      30 | chalcopyrite - 10.05
    41.1608 N 161.9273 E |             Starcity | Explored |      41 | chalcopyrite - 10.32
    33.7976 N 146.3241 W |              Kyocera | Explored |      30 | chalcopyrite - 11.09
     16.8505 N 49.0956 W |               Europa | Explored |      18 | chalcopyrite - 7.97

```
Starcity now does have a few sites on its lap that have high mining estimated concentration. 

As a matter of fact, it's actually mining this site as we speak: 
```
 41.1398 N 162.0053 E |             Starcity |    Mined |      33 | chalcopyrite - 10.07
```
It said the highest concentration is chalcopyrite and is at 10.07 %. 

So it turns out when I dragged the scroll bar down, I could indeed see chalcopyrite at 10.07% as shown below on the Mission Tool.

![image](https://user-images.githubusercontent.com/1168584/193430267-7f2ad179-6338-436b-ad4e-51441531a645.png)

**Comment by bevans2000** [comment]
A big problem is the quality of mining sites. It heavily influences the probability and the actions on site. I will change the explored console command so it's easier to created better mining sites. Then once they are easier to start via create mission command we can better explore what is going on.
In terms of the missing members, it should not be possible to embark with someone missing. So my thought is they incorrectly do a Settlement based Task during the journey which puts back in the Settlement. For Keith Watson do you have their Activity history?

**Comment by mokun** [comment]
> For Keith Watson do you have their Activity history?

Unfortunately, we only kept up to 5 days worth of activity.

On the schedule tab, I can see it keep record of activities all the way back to sol 14. 

We can no longer see what happens on Sol 8.

**Comment by mokun** [comment]
> A big problem is the quality of mining sites. It heavily influences the probability and the actions on site.

So if it comes down to this, the question would be : what real world events or activities could potentially improve the estimated quality of a mining site ?

I suppose that exploration is the major one.

And we already have the task of analyzing maps.

We could add flying drone over the site to scan the area.

May be we arrange to buy data set from other countries via the sponsor's mission control, in exchange for something...

**Comment by bevans2000** [comment]
Or increase the amount of minerals so the sites are closer to Settlements. As it stands, you have to wait a long time to mine because the sites have to be developed. So on the exploration mission we can increase the likelihood of revisiting good sites or get the sites to a good level in less visits.

**Comment by mokun** [comment]
> Or increase the amount of minerals so the sites are closer to Settlements.

Yes. we've increased # of sites having a stash of mineral recently.

But we'll continue to tweak and refine how it can be done.

**Comment by mokun** [comment]
@bevans2000 @Urwumpe 

I just committed some changes to boost up the mineral to be extracted. Before that, it seems very little or none were scooped up in each mining mission.

> Or increase the amount of minerals so the sites are closer to Settlements.

We could increase the confidence of a mining mission when it's very close to home.

As a matter of fact, we have recently intentionally generated some mineral concentration spread across the settlement and nearby areas. It may be of low concentrations. 

But I was hoping to jump start some nearby mining mission easier.

So we may want to adjust the mission planning review score when a mining mission is near.

>  As it stands, you have to wait a long time to mine because the sites have to be developed. 

Right. Let's add more contents to what it means by getting a better mineral estimation. 

It's all about prospecting. Areology skill will be handy. And perhaps paying for the mining right to a mutually trusted body of organization that handles dispute, etc.

Anyway, we can start small in modeling these things.

> So on the exploration mission we can increase the likelihood of revisiting good sites or get the sites to a good level in less visits.

Sure. We haven't talked about what instrument to bring to exploration.

**Comment by mokun** [comment]
> A big problem is the quality of mining sites. It heavily influences the probability and the actions on site

@bevans2000, @bdowrick, @Urwumpe  

Lately, we've done a lot of tweaking on estimating mineral concentrations.

Our recent work on `RandomMineralMap` [#935](https://github.com/mars-sim/mars-sim/issues/935)) includes redefining how mineral concentrations are supposed to be generated and made visible as a Mineral Map layer and how to make the code more efficient to display.

So personally I've finally got the good hang of understanding what the legacy codes were trying to model, as well as the limitations it used to have.

This old concept of mineral concentrations was all about connecting the importance of having Exploration Missions to the start of Mining Missions. 

It makes total sense at the moment for previous developers that being at a site during a Exploration Mission, the job of the settlers were to do *estimation improvement* on the mineral concentrations. The higher the number, the closer the estimation match up with the actual content.

But this was missing many critical pieces of details when it comes to the process of actually mining a mineral.

Things such as, scanning and identifying a worthy region of interest for minerals via satellite map, analyzing the morphology of the landscapes of interest to determine what minerals are there and how they came to be, etc.

Also, after personally visiting a site and using instruments to affirm the mineral contents, where's the idea of staking a claim so that others won't try to get out to mine this discovered site in a sneaky way :D ? 

In other words, on what authority and who would be the neutral organization or party that register and record the boundary of these claimed sites ?

So we really should take one step further toward this direction and envision what mining would really look like on Mars some day. 

We can base that process, at least at the beginning, to be somewhat similar to our earthly analog of what constitute a mining right for sure.

Feel free to chime in.

**Comment by mokun** [comment]
I'll just commit the changes I'm putting together to lay some groundwork for many more functional changes on mining to follow as we discuss further.
